tests/test_ble_lifecycle_controller.py

```
@@ -0,0 +1,671 @@
+    def test_resolve_hook_with_callable_hook_returns_hook(self) -> None:
+        """_resolve_hook should return the hook when it's a configured callable."""
+        # Create a mock that is NOT an unconfigured mock (has return_value set)
+        configured_hook = MagicMock(return_value=True)
+        configured_hook._mock_return_value = True  # Makes it "configured"
+        configured_hook.side_effect = None
+        configured_hook.call_args_list = []
+
+        resolved_is_closing: list[bool] = []
+
+        def _discard_invalidated_connected_client(
+            *_args: Any,
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*_args`

(ANN401)